bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt

```diff
@@ -380,6 +387,31 @@ private constructor(
         }
     }
 
+    /**
+     * Get a response from the Mastodon instance defined for this client using the PUT method.
+     * @param path an absolute path to the API endpoint to call
+     * @param body the parameters to use in the request body for this request; may be null
+     */
+    fun put(path: String, body: Parameters?): Response {
+        if (body == null) {
```

Comment:
This null check is contradictory to what is written in the Kdoc. If `null` is allowed, we should not throw an exception if it is 🙂

Reply:
Not sure what I had in mind there... :D Will have another look at it tomorrow.

Reply:
@andregasser I now removed the null check in `put()` and `patch()`.

My reason for adding them in the first place was that neither HTTP method is really sensible to perform without any actual data in the request. While that is true, checking for null is not enough, we would actually need to check that the parameter body is **not empty** on top of that - and then we'd still have cases where the body contains data, but that data is invalid for the specific endpoint that is called.

Since we control most of the callsites of these functions - and calling them with invalid arguments will still lead to an exception being thrown, only later - it is not necessary for the null check to exist there.